plugins/inputs/t128_peer_path/t128_peer_path_test.go

```diff
@@ -0,0 +1,280 @@
+	ExpectedErrors   []string
+	ExcludeHostname  bool
+	ExpectedRequests []int
+}{
+	{
+		Name:            "empty adjacent hostname",
+		ExcludeHostname: false,
+		Endpoint: Endpoint{"/api/v1/graphql/", 200, ValidPeerPathRequestWithHostName, `{
+				"data": {
+				  "allPeers": {
+					"nodes": [
+					  {
+						"routerName": "NorthEast",
+						"paths": [
+						  {
+							"node": "node1",
+							"adjacentAddress": "10.10.10.10",
+							"adjacentHostname": null,
+							"status": "UP",
+							"enabled": true,
+							"deviceInterface": "wan5",
+							"networkInterface": "wan5",
+							"vlan": "0"
+						  }
+						]
+					  }
+					]
+				  }
+				}
+			}`},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-collector",
+				Tags: map[string]string{
+					"adjacentAddress":  "10.10.10.10",
+					"deviceInterface":  "wan5",
+					"networkInterface": "wan5",
+					"node":             "node1",
+					"vlan":             "0",
+					"routerName":       "NorthEast",
+				},
+				Fields: map[string]interface{}{
+					"status":  "UP",
+					"enabled": true,
+				},
+			},
+		},
+		ExpectedErrors:   []string{},
+		ExpectedRequests: []int{1},
+	},
+	{
+		Name:            "exclude hostname set to true",
+		ExcludeHostname: true,
+		Endpoint: Endpoint{"/api/v1/graphql/", 200, ValidPeerPathRequestWithoutHostName, `{
+				"data": {
+				  "allPeers": {
+					"nodes": [
+					  {
+						"routerName": "NorthEast",
+						"paths": [
+						  {
+							"node": "node1",
+							"adjacentAddress": "10.10.10.10",
+							"adjacentHostname": "fake-peer",
+							"status": "UP",
+							"enabled": true,
+							"deviceInterface": "wan5",
+							"networkInterface": "wan5",
+							"vlan": "0"
+						  }
+						]
+					  }
+					]
+				  }
+				}
+			}`},
+		ExpectedMetrics: []*testutil.Metric{
+			{
+				Measurement: "test-collector",
+				Tags: map[string]string{
+					"adjacentAddress":  "10.10.10.10",
+					"deviceInterface":  "wan5",
+					"networkInterface": "wan5",
+					"node":             "node1",
+					"vlan":             "0",
+					"routerName":       "NorthEast",
+				},
+				Fields: map[string]interface{}{
+					"status":  "UP",
+					"enabled": true,
+				},
+			},
+		},
+		ExpectedErrors:   []string{},
+		ExpectedRequests: []int{1},
+	},
+	{
+		Name:            "multiple paths",
+		ExcludeHostname: false,
+		Endpoint: Endpoint{"/api/v1/graphql/", 200, ValidPeerPathRequestWithHostName, `{
+					"data": {
+					  "allPeers": {
+						"nodes": [
+						  {
+							"routerName": "NorthEast",
+							"paths": [
+							  {
+								"node": "node1",
+								"adjacentAddress": "172.16.3.2",
+								"adjacentHostname": null,
+								"status": "UP",
+								"enabled": true,
+								"deviceInterface": "wan5",
+								"networkInterface": "wan5",
+								"vlan": "0"
+							  },
+							  {
+	                            "node": "node1",
+	                            "adjacentAddress": "127.117.97.105",
```

Comment:
indentation is a bit weird here. Also this is to test the one filtering case. Should we add it as a seperate testcase to make it more clear?

Also about the other filtering case? Is that covered?

Reply:
yes I will add the other testcases too!